### DISCUSSION TITLE
Handle timer events

### DISCUSSION THREAD

**Discussion by rfbeiboer** [custom-components/pyscript#389] [Q&A]
Is it possible to handle timer events? Like timer.finished and timer.cancelled?
If I try @event_trigger("timer.finished"), it does not seem to work. I cannot find any documentation or examples how to do this. Is this even possible?

**Comment by idogrf** [comment]
I've been using `@event_trigger('timer.cancelled')` and `@event_trigger('timer.finished')` without issues for quite some time...
See the example bellow - 

```
@event_trigger('timer.cancelled')
def timer_cancelled(**kwargs):
    if kwargs['entity_id'] == 'timer.kitchen_timeout':
        log.info('kitchen_timeout')
        turn_lights_on()
```

What doesn't work for you? does it not trigger at all?
